src/Model/index.js

```diff
@@ -118,6 +120,28 @@ export default class Model {
     return this
   }
 
+  prepareMarkAsDeleted(): this {
+    invariant(this._isCommitted, `Cannot mark an uncomitted record as deleted`)
```

Comment:
I *think* we should also protect againt deleting a record with pending update

Reply:
I thought about this as well. In the end my conclusion was deletion should just take precedence, i.e. if a user tries to delete an updated record - does it then matter it was not synced? It should still delete, right?

Reply:
Right, but "pending update" means that someone called `prepareUpdate()` synchronously (so just before) calling `prepareMarkAsDeleted()` (not that it's pending sync or something like that). And I can't think of a good reason to batch both an update and a delete for a record — so if someone does that, it's almost surely a mistake. We can always relax this invariant, but I just fear some weird edge case showing up down the line without any test covering it

Reply:
I would suggest issuing a warning in dev mode, but still doing the delete. What do you think?

Reply:
I suggest doing an error. My reasoning: it's easier to relax this if someone complains that this breaks their workflow, then to tighten it and break people's apps if someone relies on this without a good reason